src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/configurations/TestConfigurationUtil.java

```diff
@@ -0,0 +1,116 @@
+import com.facebook.buck.intellij.ideabuck.build.BuckQueryCommandHandler;
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.google.common.base.Function;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiMethod;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public final class TestConfigurationUtil {
+
+  private static final String[] TEST_ANNOTATIONS = {
+    "org.junit.Test", "org.testng.annotations.Test"
+  };
+
+  /**
+   * Create handler that fills out test configuration using buck query on current file.
+   *
+   * @param testConfiguration the {@link TestConfiguration} to fill out.
+   * @param name a {@link String} representing the name of the configuration.
+   * @param testSelectors a {@link String} representing optional testSelectors for filtering.
+   * @param project {@link Project} then intellij project corresponding the the file or module under
+   *     test.
+   * @param containingFile {@link VirtualFile} the file that containing the impacted tests.
+   * @param buckFile {@link VirtualFile} the file representing the buck File.
+   * @param testConfigurationProcessor a {@link Function<TestConfiguration, null>} representing a
+   *     callback to run after the test configuration is filled out.
+   * @return a {@link BuckCommandHandler} that handles the buck query command used to fill out the
+   *     test configuration.
+   */
+  public static BuckCommandHandler getTestConfigurationDataHandler(
+      @Nonnull TestConfiguration testConfiguration,
+      @Nonnull String name,
+      final Optional<String> testSelectors,
+      final Project project,
+      VirtualFile containingFile,
+      VirtualFile buckFile,
+      @Nullable Function<TestConfiguration, Void> testConfigurationProcessor) {
+    BuckCommandHandler handler =
+        new BuckQueryCommandHandler(
+            project,
+            buckFile.getParent(),
+            BuckCommand.QUERY,
+            new Function<List<String>, Void>() {
+              @Nullable
+              @Override
+              public Void apply(@Nullable List<String> strings) {
+                if (!(strings == null
+                    || strings.isEmpty()
+                    || strings.get(0) == null
+                    || strings.get(0).isEmpty())) {
+                  testConfiguration.setName(name);
+                  testConfiguration.data.target = strings.get(0);
+                  testConfiguration.data.testSelectors = testSelectors.orElse("");
+                }
+                if (testConfigurationProcessor != null) {
+                  testConfigurationProcessor.apply(testConfiguration);
+                }
+                return null;
+              }
+            });
+    handler.command().addParameter("kind('java_test', owner(" + containingFile.getPath() + "))");
```

Comment:
Yeah, I think that should work. Ideally we need to have a way to indicate that we need only test targets, since there is no way right now, let's use `*_java`.